ui/src/main/js/Navigation.js

```diff
@@ -15,145 +14,90 @@ import { SCHEDULING_INFO } from 'global/components/scheduling/SchedulingModel';
 import { SETTINGS_INFO } from 'global/components/settings/SettingsModel';
 import { AUTHENTICATION_INFO } from 'global/components/auth/AuthenticationModel';
 import { BLACKDUCK_INFO } from 'global/providers/blackduck/BlackDuckModel';
+import { AUDIT_INFO } from 'global/components/audit/AuditModel';
+import { CERTIFICATE_INFO } from 'global/components/certificates/CertificateModel';
+import { TASK_MANAGEMENT_INFO } from 'global/components/task/TaskManagementModel';
+import { USER_MANAGEMENT_INFO } from 'global/components/user/UserModel';
 
-class Navigation extends Component {
-    constructor(props) {
-        super(props);
-        this.createNavItemForDescriptors = this.createNavItemForDescriptors.bind(this);
-        this.createStaticNavItem = this.createStaticNavItem.bind(this);
-    }
+const Navigation = ({ confirmLogoutPressed, descriptorMap }) => {
+    const createStaticNavItem = (uriPrefix, itemObject) => (
+        <li key={itemObject.key}>
+            <NavLink to={`${uriPrefix}${itemObject.url}`} activeClassName="activeNav">
+                {itemObject.label}
+            </NavLink>
+        </li>
+    );
 
-    createStaticNavItem(uriPrefix, itemObject) {
-        return (
-            <li key={itemObject.key}>
-                <NavLink to={`${uriPrefix}${itemObject.url}`} activeClassName="activeNav">
-                    {itemObject.label}
-                </NavLink>
-            </li>
-        );
-    }
+    const channelUri = '/alert/channels/';
+    const providerUri = '/alert/providers/';
+    const componentUri = '/alert/components/';
 
-    createNavItemForDescriptors(descriptorType, context, uriPrefix, header) {
-        const { descriptors } = this.props;
-        if (!descriptors) {
-            return null;
-        }
-        const descriptorList = DescriptorUtilities.findDescriptorByTypeAndContext(descriptors, descriptorType, context);
-        if (!descriptorList) {
-            return null;
-        }
+    const doesDescriptorExist = (key) => Object.prototype.hasOwnProperty.call(descriptorMap, key);
```

Comment:
You use this in a couple places.  In the util directory should we have a DescriptorUtils.js file where we put the construction of the descriptor map code and this check function?